src/azure-cli/azure/cli/command_modules/network/_validators.py

```diff
@@ -246,9 +246,9 @@ def validate_ssl_cert(namespace):
         if namespace.key_vault_secret_id:
             return
         # cert supplied -- use HTTPS
-        if not all(params):
```

Comment:
When generating cert-file with password, then you have to input right password, otherwise the cloud service will return error message; when generating cert-file with no password, then you needn't input password and cloud service support this kind of usage. Therefore, we need not verify the password because it will be checked on backend server.

Reply:
Could you post the error service response if a password is needed for the cert file but user don't provide it via `--cert-password`?

Reply:
BadRequestError: Password specified for certificate /subscriptions/000000000000000000/resourceGroups/myrg/providers/Microsoft.Network/applicationGateways/myag/sslCertificates/myfile is incorrect.

Reply:
The response says the password is incorrect. Is there any possibility to catch this kind of error to tell the user whether the cert file requires a password to input instead of leaving it empty or the password is incorrect even though the user has given one?

If it's not hard to implement, we could help the service team to patch this behavior. Otherwise, we could make a suggestion to service team.

Reply:
First, the cloud server only return one kind of error message and we can not distinguish the difference. Second, the user should know whether the file has password or not. And if CLI promote some specific message, it may bring some security danger.